zebra-chain/src/orchard/shielded_data.rs

```diff
@@ -38,6 +38,28 @@ pub struct ShieldedData {
     pub binding_sig: Signature<Binding>,
 }
 
+impl ShieldedData {
+    /// Iterate over the [`Action`]s for the [`AuthorizedAction`]s in this transaction.
+    pub fn actions(&self) -> impl Iterator<Item = &Action> {
+        self.actions.actions()
+    }
+
+    /// Collect the [`Nullifier`]s for this transaction, if it contains
+    /// [`Action`]s.
+    pub fn nullifiers(&self) -> impl Iterator<Item = &Nullifier> {
+        self.actions().map(|action| &action.nullifier)
+    }
+}
+
+impl AtLeastOne<AuthorizedAction> {
+    /// Iterate over the [`Action`]s of the `AuthorizedAction` for this transaction,
+    /// returning them as their generic type.
+    pub fn actions(&self) -> impl Iterator<Item = &Action> {
+        self.iter()
+            .map(|authorized_action| &authorized_action.action)
+    }
+}
+
```

Comment:
Please split these `Action` access functions into a separate PR, so we can make progress on #1980.

Reply:
This is now https://github.com/ZcashFoundation/zebra/pull/2199 , if we merge that first we need to rebase on `main` here.

Reply:
Thanks!